src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3264,6 +3264,7 @@
     <Field Name="DotToken" Type="SyntaxToken">
       <Kind Name="DotToken"/>
     </Field>
+    <Field Name="DiscardedExplicitEventName" Type="string"/>
```

Comment:
><Field Name="DiscardedExplicitEventName" Type="string"/> [](start = 4, length = 56)

I do not think this is the right approach. Either we think the name is part of the event declaration, or we don't. The syntax tree should reflect that, diagnostics should reflect what the syntax tree has. The way to fix the issue is to build a syntax tree that corresponds to our understanding where the identifier after the dot should belong. We can also make adjustments to the parser to avoid reporting additional undesired diagnostics.

Reply:
Assuming the approach works - it seems to give us the best of both worlds. The parser reports least noise, and the diagnostic message is correct. Why must we insist on the binary decision that the name either belongs or not belongs to the event declaration? I probably broke some properties that the SyntaxTree must have that I am unaware of.

Reply:
> Assuming the approach works - it seems to give us the best of both worlds. The parser reports least noise, and the diagnostic message is correct.

We are in control what parser reports, I am sure there is a way to get to the same number of errors reported.

> Why must we insist on the binary decision that the name either belongs or not belongs to the event declaration?

Because this is what defines whether the event has the name. The diagnostics is just one way to observe what the syntax tree looks like. There are other ways to do that. For example, by looking into syntax nodes directly. If we think event should have a name for a particular scenario, syntax node should reflect that. If it doesn't then there is a bug in the parser. Trying to fix diagnostics without fixing the syntax node, is simply fixing a symptom rather than a problem.

Reply:
I were assuming that the syntax tree is just an implementation detail of the compiler, which seems to be wrong since one could directly observe the syntax nodes. I agree putting a hack on a publicly observable tree seems weird and will abandon this approach.